nemo/collections/nlp/models/dialogue/intent_bio_type_classification_model.py

```diff
@@ -0,0 +1,844 @@
+        
+        dot_product_score = self.get_entity_description_score(mention_hidden_states_pad)
+        dot_product_score_log_softmax = torch.log_softmax(dot_product_score, dim=-1)
+        
+        slot_logits = self.slot_mlp(mention_hidden_states_pad)
+      
+        return intent_logits, slot_logits, bio_slot_logits, dot_product_score_log_softmax
+
+    def calculate_loss(self, intent_logits, slot_logits, bio_slot_logits, dot_product_score_log_softmax, loss_mask, intent_labels, bio_slot_labels, bio_mention_labels, mention_loss_mask):
+        # calculate combined loss for intents and slots
+        intent_loss = self.intent_loss(logits=intent_logits, labels=intent_labels)
+        bio_slot_loss = self.bio_slot_loss(logits=bio_slot_logits, labels=bio_slot_labels, loss_mask=loss_mask)
+        
+        # otherwise cross-entropy function returns error if all sentences in the batch have no entity
+        if torch.sum(mention_loss_mask).item() == 0.0:
+            slot_loss_mention_only = 0.0
+            slot_loss_mention_and_description = 0.0
+        else:
+            slot_loss_mention_only = self.slot_loss(logits=slot_logits, labels=bio_mention_labels, loss_mask=mention_loss_mask)
+            slot_loss_mention_and_description = self.slot_loss(logits=dot_product_score_log_softmax, labels=bio_mention_labels, loss_mask=mention_loss_mask)
+
+        loss = self.total_loss(loss_1=intent_loss, loss_2=bio_slot_loss, loss_3=slot_loss_mention_only, loss_4=slot_loss_mention_and_description)
+        return loss
+
+    def training_step(self, batch, batch_idx):
+        """
+        Lightning calls this inside the training loop with the data from the training dataloader
+        passed in as `batch`.
+        """
+        # forward pass
+        input_ids, input_type_ids, input_mask, loss_mask, subtokens_mask, intent_labels, slot_labels, bio_slot_labels, bio_mention_labels, mention_loss_mask = batch
+        intent_logits, slot_logits, bio_slot_logits, dot_product_score_log_softmax = self(
+            input_ids=input_ids, token_type_ids=input_type_ids, attention_mask=input_mask, mention_mask=bio_slot_labels
+        )
+
+        train_loss = self.calculate_loss(intent_logits, slot_logits, bio_slot_logits, dot_product_score_log_softmax, loss_mask, intent_labels, bio_slot_labels, bio_mention_labels, mention_loss_mask)
+                        
+        lr = self._optimizer.param_groups[0]['lr']
+
+        self.log('train_loss', train_loss)
+        self.log('lr', lr, prog_bar=True)
+
+        return {
+            'loss': train_loss,
+            'lr': lr,
+        }
+
+    def validation_step(self, batch, batch_idx):
+        """
+        Lightning calls this inside the validation loop with the data from the validation dataloader
+        passed in as `batch`.
+        """
+        input_ids, input_type_ids, input_mask, loss_mask, subtokens_mask, intent_labels, slot_labels, bio_slot_labels, bio_mention_labels, mention_loss_mask = batch
```

Comment:
One final thing before you add new files next time is to use `python setup.py style --fix` to lint the code so that overly long lines can be cut into max length of 80 characters. Once you do this, you can write the Jenkins CI and set the status of this PR to 'ready for review'.